codec-http2/src/main/java/io/netty/handler/codec/http2/Http2FrameCodec.java

```diff
@@ -146,11 +226,49 @@ public Http2FrameCodec(boolean server, Http2FrameLogger frameLogger) {
         decoder.frameListener(new FrameListener());
         http2Handler = new InternalHttp2ConnectionHandler(decoder, encoder, initialSettings);
         http2Handler.connection().addListener(new ConnectionListener());
+        streamKey = connection().newKey();
         this.server = server;
     }
 
-    Http2ConnectionHandler connectionHandler() {
-        return http2Handler;
+    // TODO(buchgr): Discuss: Should this method be thread safe?
+    Http2FrameStream newStream() {
+        ChannelHandlerContext ctx0 = ctx;
+        if (ctx0 == null) {
+            throw new IllegalStateException("Channel handler not added to a channel pipeline.");
+        }
+
+        return new Http2Stream2Impl(ctx0.channel());
+    }
+
+    /**
+     * Iterates over all active HTTP/2 streams.
+     *
+     * <p>This method must not be called outside of the event loop.
+     */
+    void forEachActiveStream(final Http2FrameStreamVisitor streamVisitor) throws Http2Exception {
+        assert ctx.channel().eventLoop().inEventLoop();
+
+        connection().forEachActiveStream(new Http2StreamVisitor() {
+            @Override
+            public boolean visit(Http2Stream stream) {
+                Http2FrameStream stream2 = stream.getProperty(streamKey);
+                if (stream2 == null) {
+                    /**
+                     * This code is expected to almost never execute. However, in rare cases it's possible that a
+                     * stream is active without a {@link Http2FrameStream} object attached, as it's set in a listener of
+                     * the HEADERS frame write.
+                     */
+                    // TODO(buchgr): Remove once Http2Stream2 and Http2Stream are merged.
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1135)